Add support for entity_view

steveworley opened the pull request:
`entity_view` was [deprecated in 8.0.0](https://api.drupal.org/api/drupal/core%21includes%21entity.inc/function/entity_view/8.8.x).

Some complexities with the change in API as moving to this no longer supports the `$reset` parameter, but this will convert `entity_view` to a Drupal 9 compatible expression.

ccampos committed:
Update README.md

A couple of typos

steveworley committed:
Add entity view support.

steveworley committed:
Remove support for cache clear.

damontgomery commented on rector_examples/entity_view.php:
```diff
@@ -0,0 +1,33 @@
+function lagncode_example() {
+  $entity = new stdClass();
+  $langcode = 'de';
+  $veiw = entity_view($entity, 'default', $langcode);
```
minor: spelling of `$view`.

damontgomery commented on rector_examples/entity_view.php:
```diff
@@ -0,0 +1,33 @@
+}
+
+function reset_cache_example() {
+  // Not supported.
```
Thanks for calling this out.

damontgomery commented on src/Rector/Deprecation/Base/EntityViewBase.php:
```diff
@@ -0,0 +1,76 @@
+
+    $name = new Node\Name\FullyQualified('Drupal');
+
+    $entity_type_method_name = new Node\Identifier('entityTypeManager');
```
Swap this for `$entityTypeManager_method_name`? like the ones below?

damontgomery commented on src/Rector/Deprecation/Base/EntityViewBase.php:
```diff
@@ -0,0 +1,76 @@
+
+    $getViewBuilder_method_name = new Node\Identifier('getViewBuilder');
+
+    $entityRef = $node->args[0]->value;
```
Could we make this `$entity_reference`? I like how the names you've used are mostly matching the method names like `$getViewBuilder_method_name` which is very clear to me. :)

damontgomery commented on src/Rector/Deprecation/Base/EntityViewBase.php:
```diff
@@ -0,0 +1,76 @@
+use Rector\Core\Rector\AbstractRector;
+
+/**
+ * Replaced deprecated entity_load() calls.
```
Can we combine `EntityViewBase.php` and `EntityViewRector`? It looks like there was a base class, which we could refactor later, but for now, putting them together makes more sense to me while reading. The comments seem like they need to be updated too.

damontgomery commented on src/Rector/Deprecation/Base/EntityViewBase.php:
```diff
@@ -0,0 +1,76 @@
+abstract class EntityViewBase extends AbstractRector
+{
+
+  use AddCommentTrait;
```
Did you want to add a comment to the replacement about checking for the `reset` option?

If not, we could remove this trait.

damontgomery reviewed (changes requested):
Thanks, @steveworley! And, hello!

I left a few comments. This looks close.

There are three other requirements for adding new rules
- Adding the new rector rule to the configuration file, config/drupal-8/drupal-8.0-deprecations.yml
- Updating the index (deprecation-index.yml)
- Adding the after example like rector_examples_updated/entity_create.php.

Also, there is a fairly large refactor going on to support newer versions of Rector. Because of that, I'm hesitant to merge new rules over the next week or so. If you want to build off that branch, we're in the final stages of moving in that direction. #134. This code should mostly work, but the configuration files have moved from YAML to php and some of the `use` statements have changed. Example: https://github.com/palantirnet/drupal-rector/pull/134/files#diff-069c02b1633ca197bab8d62713679cae3a5a80118c3b0a4ee51100c749ae6360.

Sorry again for the pieces moving from under you and happy to help take a look or make the updates after that PR is merged if you don't want to or don't have the time to make the updates.

steveworley committed:
Apply review updates.

- Add to deprecation list
- Add to derecation index
- Add transformation example
- Correct spelling
- Remove base class

steveworley committed:
Update for rector 0.10.

steveworley committed:
Fix tests.

yusufhm committed:
Rename entiy_view.php to entity_view.php

yusufhm committed:
Updated comment to match.

damontgomery commented on src/Rector/Deprecation/EntityViewRector.php:
```diff
@@ -0,0 +1,95 @@
+
+        $name = new Node\Name\FullyQualified('Drupal');
+
+        $entityTypManager = new Node\Identifier('entityTypeManager');
```
minor: spelling of `$entityTypeManager` missing the `e` in `Type`.

damontgomery reviewed (approved):
Thanks for the updates. The code is clear! The automated tests pass.

I would be happy to have this merged. Not sure if we should wait until / if we merge `rector-0.10-minimal-changes`.
